packages/web-react/src/components/ScrollView/useScrollPosition.ts

```diff
@@ -0,0 +1,82 @@
+import { UIEvent, MutableRefObject, useState, useEffect } from 'react';
+import { debounce } from '../../utils';
+import { ScrollViewDirectionType } from '../../types';
+import { EDGE_DETECTION_INACCURACY_PX, DEBOUNCE_DELAY } from './const';
+
+export interface UseScrollPositionProps {
+  viewportReference: MutableRefObject<HTMLDivElement | null>;
+  contentReference: MutableRefObject<HTMLDivElement | null>;
+  direction?: ScrollViewDirectionType;
+}
+
+export interface UseScrollPositionReturn {
+  isScrolledAtEnd: boolean;
+  isScrolledAtStart: boolean;
+  onScroll: (event: UIEvent) => void;
+}
+
+export interface CurrentPositionProps {
+  bottom: number;
+  left: number;
+  right: number;
+  top: number;
+}
+
+export const useScrollPosition = ({
+  viewportReference,
+  contentReference,
+  direction,
+}: UseScrollPositionProps): UseScrollPositionReturn => {
+  const [isScrolledAtStart, setScrolledStart] = useState<boolean>(false);
+  const [isScrolledAtEnd, setScrolledAtEnd] = useState<boolean>(false);
+
+  const getElementsPositionDifference = (): CurrentPositionProps | null => {
+    if (!(contentReference.current && viewportReference.current)) {
+      return null;
+    }
+
+    const contentPosition: DOMRect = contentReference.current.getBoundingClientRect();
+    const viewportPosition: DOMRect = viewportReference.current.getBoundingClientRect();
+
+    return {
+      bottom: contentPosition.bottom - viewportPosition.bottom,
+      left: contentPosition.left - viewportPosition.left,
+      right: contentPosition.right - viewportPosition.right,
+      top: contentPosition.top - viewportPosition.top,
+    };
+  };
+
+  const handleScrollViewState = () => {
+    const isDirectionHorizontal = direction === 'horizontal';
+    const scrollPositionStart = isDirectionHorizontal ? 'left' : 'top';
```

Comment:
I think this is the same as I wrote a comment here: https://github.com/lmc-eu/spirit-design-system/pull/844/files#r1203601435

It would be great if we can use dictionaries here, like `Position.LEFT` instead of just `left`. We can then use this dictionary to define the keys of the current position object and so on.

So there should be two dictionaries, one with directions and one with alignments.

Reply:
done